README.txt

Feel free to add some documentation or simply add a link to the online manual.


in subpart <!-- ###ARTICLE_VIEW### --> you can use now this


ChangeLog
0.1.0	* alt-text, title-text and caption added

0.0.5	* Bug #8867 reparied

0.0.4	* Marker for teaser images changed. From now the marker ###PRODUCT_TEASERIMAGES### is used instead of ###PRODUCT_TEASER###. Plz. update your templates. 

0.0.3	* Marker ###PRODUCT_MANUFACTURERLOGO### added
		* doNotRender option added
		* Bug repaired by supplier and article images

0.0.2	* Markers ###ARTICLE_SUPPLIERLOGO### and ###ARTICLE_IMAGES### added
		* some PHP changes

0.0.1	* Initial release



ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext: "dam_commerce"
#
# Auto generated 11-08-2008 21:31
#
# Manual updates:
# Only the data in the array - anything else is removed by next write.
# "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Commerce DAM connector',
	'description' => 'Extends the commerce extension with the possibility to use DAM for images.',
	'category' => 'fe',
	'shy' => 0,
	'version' => '0.1.0',
	'dependencies' => '',
	'conflicts' => '',
	'priority' => '',
	'loadOrder' => '',
	'module' => '',
	'state' => 'alpha',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => 'tx_commerce_categories,tx_commerce_products,tx_commerce_articles,tx_commerce_attributes,tx_commerce_manufacturer,tx_commerce_supplier,tx_commerce_attribute_values',
	'clearcacheonload' => 0,
	'lockType' => '',
	'author' => 'Juraj Sulek',
	'author_email' => 'juraj@sulek.sk',
	'author_company' => '',
	'CGLcompliance' => '',
	'CGLcompliance_note' => '',
	'constraints' => array(
		'depends' => array(
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:16:{s:9:"ChangeLog";s:4:"090a";s:10:"README.txt";s:4:"3e9a";s:33:"class.tx_damcommerce_articles.php";s:4:"b88e";s:35:"class.tx_damcommerce_categories.php";s:4:"7623";s:33:"class.tx_damcommerce_helpfunc.php";s:4:"8586";s:33:"class.tx_damcommerce_products.php";s:4:"dda9";s:21:"ext_conf_template.txt";s:4:"248d";s:12:"ext_icon.gif";s:4:"3dec";s:17:"ext_localconf.php";s:4:"f321";s:14:"ext_tables.php";s:4:"a257";s:14:"ext_tables.sql";s:4:"31f5";s:16:"locallang_db.xml";s:4:"8c98";s:14:"doc/manual.sxw";s:4:"a0aa";s:19:"doc/wizard_form.dat";s:4:"ee90";s:20:"doc/wizard_form.html";s:4:"635e";s:16:"static/setup.txt";s:4:"fc3f";}',
	'suggests' => array(
	),
);

?>

ext_tables.php
<?php
if (!defined ('TYPO3_MODE')) 	die ('Access denied.');
$tempSetup = unserialize($TYPO3_CONF_VARS['EXT']['extConf']['dam_commerce']);
/* ------------------------------------------------------------------------------------------- */
/* categories */
t3lib_div::loadTCA('tx_commerce_categories');
$tempColumns = array (
	'tx_damcommerce_categories_images' => txdam_getMediaTCA('image_field', 'dam_commerce_cat_1')
);
$tempColumns['tx_damcommerce_categories_images']['label']=$TCA['tx_commerce_categories']['columns']['images']['label'];
t3lib_extMgm::addTCAcolumns('tx_commerce_categories',$tempColumns,1);

if ($tempSetup['add_orig_field_category_images']) {
	t3lib_extMgm::addToAllTCAtypes('tx_commerce_categories','tx_damcommerce_categories_images','','after:images');
} else {
	$TCA['tx_commerce_categories']['types']['0']['showitem'] = str_replace(', images,', ', tx_damcommerce_categories_images,', $TCA['tx_commerce_categories']['types']['0']['showitem']);
}
/* ------------------------------------------------------------------------------------------- */
/* products */
t3lib_div::loadTCA('tx_commerce_products');
$tempColumns_2 = array (
	'tx_damcommerce_prod_images' => txdam_getMediaTCA('image_field', 'dam_commerce_prod_1'),
	'tx_damcommerce_prod_teaserimages' => txdam_getMediaTCA('image_field', 'dam_commerce_prod_2')
);
$tempColumns_2['tx_damcommerce_prod_images']['label']=$TCA['tx_commerce_products']['columns']['images']['label'];
$tempColumns_2['tx_damcommerce_prod_teaserimages']['label']=$TCA['tx_commerce_products']['columns']['teaserimages']['label'];
t3lib_extMgm::addTCAcolumns('tx_commerce_products',$tempColumns_2,1);

if ($tempSetup['add_orig_field_product_images']) {
	t3lib_extMgm::addToAllTCAtypes('tx_commerce_products','tx_damcommerce_prod_images','','after:images');
} else {
	$TCA['tx_commerce_products']['types']['0']['showitem'] = str_replace(', images,', ', tx_damcommerce_prod_images,', $TCA['tx_commerce_products']['types']['0']['showitem']);
}

if ($tempSetup['add_orig_field_product_teaserimages']) {
	t3lib_extMgm::addToAllTCAtypes('tx_commerce_products','tx_damcommerce_prod_teaserimages','','after:teaserimages');
} else {
	$TCA['tx_commerce_products']['types']['0']['showitem'] = str_replace(', teaserimages,', ', tx_damcommerce_prod_teaserimages,', $TCA['tx_commerce_products']['types']['0']['showitem']);
}
/* ------------------------------------------------------------------------------------------- */
/* articles */
t3lib_div::loadTCA('tx_commerce_articles');
$tempColumns_3 = array (
	'tx_damcommerce_art_images' => txdam_getMediaTCA('image_field', 'dam_commerce_art_1')
);
$tempColumns_3['tx_damcommerce_art_images']['label']=$TCA['tx_commerce_articles']['columns']['images']['label'];
t3lib_extMgm::addTCAcolumns('tx_commerce_articles',$tempColumns_3,1);

if ($tempSetup['add_orig_field_articles_images']) {
	t3lib_extMgm::addToAllTCAtypes('tx_commerce_articles','tx_damcommerce_art_images','','after:images');
} else {
	$TCA['tx_commerce_articles']['types']['0']['showitem'] = str_replace('images', 'tx_damcommerce_art_images', $TCA['tx_commerce_articles']['types']['0']['showitem']);
}
/* ------------------------------------------------------------------------------------------- */
/* atributes */
t3lib_div::loadTCA('tx_commerce_attributes');
$tempColumns_4 = array (
	'tx_damcommerce_atrib_icon' => txdam_getMediaTCA('image_field', 'dam_commerce_atr_1')
);
$tempColumns_4['tx_damcommerce_atrib_icon']['label']=$TCA['tx_commerce_attributes']['columns']['icon']['label'];
$tempColumns_4['tx_damcommerce_atrib_icon']['config']['maxitems']=1;
$tempColumns_4['tx_damcommerce_atrib_icon']['config']['size']=2;
t3lib_extMgm::addTCAcolumns('tx_commerce_attributes',$tempColumns_4,1);

if ($tempSetup['add_orig_field_atributes_icon']) {
	t3lib_extMgm::addToAllTCAtypes('tx_commerce_attributes','tx_damcommerce_atrib_icon','','after:icon');
} else {
	$TCA['tx_commerce_attributes']['types']['0']['showitem'] = str_replace(', icon;', ', tx_damcommerce_atrib_icon;', $TCA['tx_commerce_attributes']['types']['0']['showitem']);
}
/* ------------------------------------------------------------------------------------------- */
/* tx_commerce_manufacturer */
t3lib_div::loadTCA('tx_commerce_manufacturer');
$tempColumns_5 = array (
	'tx_damcommerce_manufac_logo' => txdam_getMediaTCA('image_field', 'dam_commerce_man_1')
);
$tempColumns_5['tx_damcommerce_manufac_logo']['label']=$TCA['tx_commerce_manufacturer']['columns']['logo']['label'];
$tempColumns_5['tx_damcommerce_manufac_logo']['config']['maxitems']=1;
$tempColumns_5['tx_damcommerce_manufac_logo']['config']['size']=2;
t3lib_extMgm::addTCAcolumns('tx_commerce_manufacturer',$tempColumns_5,1);

if ($tempSetup['add_orig_field_manufacturers_logo']) {
	t3lib_extMgm::addToAllTCAtypes('tx_commerce_manufacturer','tx_damcommerce_manufac_logo','','after:logo');
} else {
	$TCA['tx_commerce_manufacturer']['types']['0']['showitem'] = str_replace('logo;', 'tx_damcommerce_manufac_logo;', $TCA['tx_commerce_manufacturer']['types']['0']['showitem']);
}
/* ------------------------------------------------------------------------------------------- */
/* tx_commerce_supplier */
t3lib_div::loadTCA('tx_commerce_supplier');
$tempColumns_6 = array (
	'tx_damcommerce_supl_logo' => txdam_getMediaTCA('image_field', 'dam_commerce_sup_1')
);
$tempColumns_6['tx_damcommerce_supl_logo']['label']=$TCA['tx_commerce_supplier']['columns']['logo']['label'];
$tempColumns_6['tx_damcommerce_supl_logo']['config']['maxitems']=1;
$tempColumns_6['tx_damcommerce_supl_logo']['config']['size']=2;
t3lib_extMgm::addTCAcolumns('tx_commerce_supplier',$tempColumns_6,1);

if ($tempSetup['add_orig_field_suplier_logo']) {
	t3lib_extMgm::addToAllTCAtypes('tx_commerce_supplier','tx_damcommerce_supl_logo','','after:logo');
} else {
	$TCA['tx_commerce_supplier']['types']['0']['showitem'] = str_replace('logo;', 'tx_damcommerce_supl_logo;', $TCA['tx_commerce_supplier']['types']['0']['showitem']);
}
/* ------------------------------------------------------------------------------------------- */
/* tx_commerce_attribute_values */
t3lib_div::loadTCA('tx_commerce_attribute_values');
$tempColumns_7 = array (
	'tx_damcommerce_atr_val_icon' => txdam_getMediaTCA('image_field', 'dam_commerce_atrval_1')
);
$tempColumns_7['tx_damcommerce_atr_val_icon']['label']=$TCA['tx_commerce_attribute_values']['columns']['icon']['label'];
$tempColumns_7['tx_damcommerce_atr_val_icon']['config']['maxitems']=1;
$tempColumns_7['tx_damcommerce_atr_val_icon']['config']['size']=2;
t3lib_extMgm::addTCAcolumns('tx_commerce_attribute_values',$tempColumns_7,1);

if ($tempSetup['add_orig_field_atrval_icon']) {
	t3lib_extMgm::addToAllTCAtypes('tx_commerce_attribute_values','tx_damcommerce_atr_val_icon','','after:icon');
} else {
	$TCA['tx_commerce_attribute_values']['types']['0']['showitem'] = str_replace(', icon,', ', tx_damcommerce_atr_val_icon,', $TCA['tx_commerce_attribute_values']['types']['0']['showitem']);
}

/* ------------------------------------------------------------------------------------------- */
/* static template */

t3lib_extMgm::addStaticFile($_EXTKEY,'static/','DAM COMMERCE');
?>

class.tx_damcommerce_articles.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2007 Juraj Sulek (juraj@sulek.sk)
*  All rights reserved
*
*  This script is part of the Typo3 project. The Typo3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * Part of the "Commerce DAM connector" extension.
 *
 * @author	Juraj Sulek <juraj@sulek.sk>
 * @package commerce DAM connector
 * @subpackage
 */
/**
 * [CLASS/FUNCTION INDEX of SCRIPT]
 *
 *
 *
 *   45: class tx_damcommerce_articles
 *   55:     function additionalMarkerArticle($markerArray,$element,$this_default)
 *
 * TOTAL FUNCTIONS: 1
 * (This index is automatically created/updated by the extension "extdeveval")
 *
 */
require_once(t3lib_extMgm::extPath('dam_commerce').'class.tx_damcommerce_helpfunc.php');

class tx_damcommerce_articles{

	/**
	 * Add DAM elements to aricles-markers
	 *
	 * @param	array		$markerArray: array with processed markers
	 * @param	object		$element: object with data about article
	 * @param	object		$this_default: this object
	 * @return	array		a array with filled DAM markers
	 */
	function additionalMarkerArticle($markerArray,$element,$this_default){
		/* categories - list */
		$uid=$element->_LOCALIZED_UID ? $element->_LOCALIZED_UID : $element->uid;
		$tempSetup = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['dam_commerce']);
		/* article images */
		if($this_default->conf['listView.']['articles.']['fields.']['DAM_images.']['disable']!=1){
			if(!$tempSetup['add_orig_field_articles_images']) {
				$markerArray['ARTICLE_IMAGES']='';
			};
			tx_damcommerce_helpfunc::getImage($markerArray['ARTICLE_IMAGES'],$this_default,$this_default->conf['listView.']['articles.']['fields.']['DAM_images.'],$uid,'tx_commerce_articles','dam_commerce_art_1');
		}
		
		/* supplier logo */
		if($this_default->conf['listView.']['suppliers.']['fields.']['DAM_logo.']['disable']!=1){
			$markerArray['ARTICLE_SUPPLIERLOGO'].='';
			if($element->getSupplierUid()!=0){
				//supplier icon
				if(!$tempSetup['add_orig_field_suplier_logo']) {
					$markerArray['ARTICLE_SUPPLIERLOGO']='';
				};
				tx_damcommerce_helpfunc::getImage($markerArray['ARTICLE_SUPPLIERLOGO'],$this_default,$this_default->conf['listView.']['suppliers.']['fields.']['DAM_logo.'],$element->getSupplierUid(),'tx_commerce_supplier','dam_commerce_sup_1');
			}	
		}	
		return $markerArray;
	}
}

if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/dam_commerce/class.tx_damcommerce_articles.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/dam_commerce/class.tx_damcommerce_articles.php']);
}

?>

class.tx_damcommerce_helpfunc.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2007 Juraj Sulek (juraj@sulek.sk)
*  All rights reserved
*
*  This script is part of the Typo3 project. The Typo3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * Part of the "Commerce DAM connector" extension.
 *
 * @author	Juraj Sulek <juraj@sulek.sk>
 * @package commerce DAM connector
 * @subpackage
 */
/**
 * [CLASS/FUNCTION INDEX of SCRIPT]
 *
 *
 *
 *   43: class tx_damcommerce_helpfunc
 *   55:     function function getImage(&$marker,$thisDefault,$confArray,$uid,$table,$ident)
 *
 * TOTAL FUNCTIONS: 1
 * (This index is automatically created/updated by the extension "extdeveval")
 *
 */
class tx_damcommerce_helpfunc{
	/**
	 * get the IMG tag
	 *
	 * @param	string		marker which should be filled
	 * @param	object		$this object
	 * @param	array		typoscript configuration array for the image
	 * @param	int			element uid
	 * @param	string		database table
	 * @param	string		dam ident
	 * @return	string		IMG tag
	 */
	function getImage(&$marker,$thisDefault,$confArray,$uid,$table,$ident){
		$files=tx_dam_db::getReferencedFiles($table, $uid, $ident,'tx_dam_mm_ref','tx_dam.uid,tx_dam.'.$confArray['title.']['field'].',tx_dam.'.$confArray['alt.']['field'].',tx_dam.'.$confArray['caption.']['field'].',tx_dam.file_name,tx_dam.file_dl_name,tx_dam.file_path');
		if(count($files['rows'])>0){
			foreach($files['rows'] as $key=>$val){
				$damField=$confArray;
				$damField['file']=$val['file_path'].$val['file_name'];
				$damField['altText']=$val[$confArray['alt.']['field']];
				$damField['titleText']=$val[$confArray['title.']['field']];
				$image=$thisDefault->cObj->IMAGE($damField);
				if($confArray['caption']==1){
					$image=str_replace('###IMAGE###',$image,$confArray['caption.']['layout']);
					$image=str_replace('###CAPTION###',$val[$confArray['caption.']['field']],$image);
				}
				$marker.=$image;
			}
		}		
		return $markerArray;
	}
}

if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/dam_commerce/class.tx_damcommerce_helpfunc.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/dam_commerce/class.tx_damcommerce_helpfunc.php']);
}

?>

class.tx_damcommerce_categories.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2007 Juraj Sulek (juraj@sulek.sk)
*  All rights reserved
*
*  This script is part of the Typo3 project. The Typo3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * Part of the "Commerce DAM connector" extension.
 *
 * @author	Juraj Sulek <juraj@sulek.sk>
 * @package commerce DAM connector
 * @subpackage
 */
/**
 * [CLASS/FUNCTION INDEX of SCRIPT]
 *
 *
 *
 *   45: class tx_damcommerce_categories
 *   55:     function additionalMarkerElement($markerArray,$element,$this_default)
 *
 * TOTAL FUNCTIONS: 1
 * (This index is automatically created/updated by the extension "extdeveval")
 *
 */
require_once(t3lib_extMgm::extPath('dam_commerce').'class.tx_damcommerce_helpfunc.php');

class tx_damcommerce_categories{

	/**
	 * Add DAM elements to category-markers
	 *
	 * @param	array		$markerArray: array with processed markers
	 * @param	object		$element: object with data about category
	 * @param	object		$this_default: this object
	 * @return	array		a array with filled DAM markers
	 */
	function additionalMarkerElement($markerArray,$element,$this_default){
		/* categories - list */
		$uid=$element->_LOCALIZED_UID ? $element->_LOCALIZED_UID : $element->uid;
		$tempSetup = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['dam_commerce']);
		/* category images */
		if($this_default->conf['categoryListView.']['categories.']['fields.']['DAM_images.']['disable']!=1){
			if(!$tempSetup['add_orig_field_category_images']) {
				$markerArray['IMAGES']='';
			};
			tx_damcommerce_helpfunc::getImage($markerArray['IMAGES'],$this_default,$this_default->conf['categoryListView.']['categories.']['fields.']['DAM_images.'],$uid,'tx_commerce_categories','dam_commerce_cat_1');
		}
		return $markerArray;
	}
}

if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/dam_commerce/class.tx_damcommerce_categories.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/dam_commerce/class.tx_damcommerce_categories.php']);
}

?>

class.tx_damcommerce_products.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2007 Juraj Sulek (juraj@sulek.sk)
*  All rights reserved
*
*  This script is part of the Typo3 project. The Typo3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * Part of the "Commerce DAM connector" extension.
 *
 * @author	Juraj Sulek <juraj@sulek.sk>
 * @package commerce DAM connector
 * @subpackage
 */
/**
 * [CLASS/FUNCTION INDEX of SCRIPT]
 *
 *
 *
 *   45: class tx_damcommerce_products
 *   55:     function postProcessLinkArray($markerArray,$element,$this_default)
 *
 * TOTAL FUNCTIONS: 1
 * (This index is automatically created/updated by the extension "extdeveval")
 *
 */
require_once(t3lib_extMgm::extPath('dam_commerce').'class.tx_damcommerce_helpfunc.php');

class tx_damcommerce_products{

	/**
	 * Add DAM elements to products-markers
	 *
	 * @param	array		$markerArray: array with processed markers
	 * @param	object		$element: object with data about product
	 * @param	object		$this_default: this object
	 * @return	array		a array with filled DAM markers
	 */
	function postProcessLinkArray($markerArray,$element,$this_default){
		/* products */
		$uid=$element->_LOCALIZED_UID ? $element->_LOCALIZED_UID : $element->uid;
		$tempSetup = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['dam_commerce']);
		
		/* product images */
		if($this_default->conf['listView.']['products.']['fields.']['DAM_images.']['disable']!=1){
			if(!$tempSetup['add_orig_field_product_images']) {
				$markerArray['###PRODUCT_IMAGES###']='';
			};
			tx_damcommerce_helpfunc::getImage($markerArray['###PRODUCT_IMAGES###'],$this_default,$this_default->conf['listView.']['products.']['fields.']['DAM_images.'],$uid,'tx_commerce_products','dam_commerce_prod_1');
		}
		
		/* product teaser images */
		if($this_default->conf['listView.']['products.']['fields.']['DAM_teaserimages.']['disable']!=1){
			if(!$tempSetup['add_orig_field_product_teaserimages']) {
				$markerArray['###PRODUCT_TEASERIMAGES###']='';
			};
			tx_damcommerce_helpfunc::getImage($markerArray['###PRODUCT_TEASERIMAGES###'],$this_default,$this_default->conf['listView.']['products.']['fields.']['DAM_teaserimages.'],$uid,'tx_commerce_products','dam_commerce_prod_2');
		}
		
		/* product manufacturers logo */
		if($this_default->conf['listView.']['manufacturers.']['fields.']['DAM_logo.']['disable']!=1){
			//if(!$tempSetup['add_orig_field_manufacturers_logo']) {
				$markerArray['###PRODUCT_MANUFACTURERLOGO###']='';
			//};
			
			tx_damcommerce_helpfunc::getImage($markerArray['###PRODUCT_MANUFACTURERLOGO###'],$this_default,$this_default->conf['listView.']['manufacturers.']['fields.']['DAM_logo.'],$element->manufacturer_uid,'tx_commerce_manufacturer','dam_commerce_man_1');
		}
		return $markerArray;
	}
}

if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/dam_commerce/class.tx_damcommerce_products.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/dam_commerce/class.tx_damcommerce_products.php']);
}

?>